Library/Homebrew/dev-cmd/release.rb

```diff
@@ -0,0 +1,119 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  extend T::Sig
+
+  module_function
+
+  sig { returns(CLI::Parser) }
+  def release_args
+    Homebrew::CLI::Parser.new do
+      description <<~EOS
+        Automate the Homebrew release process. Running `brew release` will
+        create a new tag, push the tag, and create a new release with release notes.
+
+        The command will check that a release is being made on the latest commit of
+        the master branch. This can be overridden with the `--force` flag.
+
+        By default, `brew release` will bump the patch version number. Pass
+        `--major` or `--minor` to bump the major or minor version numbers, respectively.
+        The command will fail if the previous major or minor release was made less than
+        one month ago. This can be overridden with the `--force` flag.
+
+        Requires write access to the Homebrew/brew repository.
+      EOS
+      switch "--major",
+             description: "Create a major release."
+      switch "--minor",
+             description: "Create a minor release."
+      switch "--force-major-minor",
+             description: "Create a release despite the warning that the latest " \
+                          "major/minor release was made less than one month ago."
+      switch "--force-mismatched-head",
+             description: "Create a release despite the warning that HEAD does not match origin/HEAD."
+      switch "--no-push",
+             description: "Create the new tag locally but do not push to the remote."
+      switch "--no-browse",
+             description: "Print the release URL instead of opening in a browser."
+      conflicts "--major", "--minor"
+
+      named_args :none
+    end
+  end
+
+  def release
+    args = release_args.parse
+
+    safe_system "git", "-C", HOMEBREW_REPOSITORY, "fetch", "origin"
```

Comment:
Could perhaps instead make this command one that always runs `brew update` before.

Reply:
Yeah, I like this better (although then it could be overridden by `HOMEBREW_NO_AUTO_UPDATE`. One question, though. How do I do that? I've never figured out how those commands to that 😅

Reply:
Ah, nevermind. This may not even be necessary if we query the GitHub API for the latest tag anyway. I'll need to play around more I think

Reply:
Agreed probably no longer necessary but: to answer your question: https://github.com/Homebrew/brew/blob/60325a6f739a3c476804b5acc432c9f64a484279/Library/Homebrew/brew.sh#L162-L177